airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -200,6 +201,10 @@ def _resolve_connection(self):
                 self._conn_id, conn_data['master']
             )
 
+            raise AirflowException("Could not load connection string %s, defaulting to %s",
```

Comment:
This message doesn't seem to match the behavoiur. It says "defaulting" but it's raising an error, so it can't continue.